src/System.Collections.Specialized/src/System/Collections/Specialized/HybridDictionary.cs

```diff
@@ -97,7 +97,7 @@ public HybridDictionary(int initialSize, bool caseInsensitive)
                     if (list.Count >= CutoverPoint - 1)
                     {
                         ChangeOver();
-                        hashtable[key] = value;
+                        hashtable![key] = value;
```

Comment:
Wouldn't this null ref? This is an else if, so the only way to get to this branch is if hashtable was null and list wasn't.

Reply:
Ah nevermind, I see, ChangeOver initializing it. Could you instead add a `Debug.Assert(hashtable != null)` that gives us more coverage in Debug mode.

Reply:
I could if that is OK/adhere with below statement:
`Each PR should contain only changes related to the nullable annotations/attributes, no other changes.`

Reply:
Debug.Assert is relevant to nullable annotations.  It doesn't affect the IL in release builds, and it's one of the ways in which we suppress nullable warnings, as an alternative to using `!`.

Reply:
`Debug.Assert()` affects nullable data flow analysis, and it is general goodness as it gives us more coverage. A `Debug.Assert` statement is not changing the product code at all. 

We used it a lot when annotating S.P.CoreLib, because of that, it is better to add that than a `!` when possible and when it makes sense to do so. Like this scenario where you're calling a helper method and it should always assign that value ensuring it will not be null.

Reply:
Got it, thanks